presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,513 @@
+    }
+
+    @Description("Returns the first point of a LINESTRING geometry as a Point")
+    @ScalarFunction("ST_StartPoint")
+    @SqlType(NAME)
+    public static Slice stStartPoint(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_StartPoint", LINE_STRING);
+        MultiPath lines = (MultiPath) geometry.getEsriGeometry();
+        SpatialReference reference = geometry.getEsriSpatialReference();
+        return serialize(OGCGeometry.createFromEsriGeometry(lines.getPoint(0), reference));
+    }
+
+    @Description("Returns the last point of a LINESTRING geometry as a Point")
+    @ScalarFunction("ST_EndPoint")
+    @SqlType(NAME)
+    public static Slice stEndPoint(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_EndPoint", LINE_STRING);
+        MultiPath lines = (MultiPath) geometry.getEsriGeometry();
+        SpatialReference reference = geometry.getEsriSpatialReference();
+        return serialize(OGCGeometry.createFromEsriGeometry(lines.getPoint(lines.getPointCount() - 1), reference));
+    }
+
+    @Description("Return the X coordinate of the point")
+    @ScalarFunction("ST_X")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stX(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_X", POINT);
+        return ((OGCPoint) geometry).X();
+    }
+
+    @Description("Return the Y coordinate of the point")
+    @ScalarFunction("ST_Y")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stY(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_Y", POINT);
+        return ((OGCPoint) geometry).Y();
+    }
+
+    @Description("Returns the closure of the combinatorial boundary of this Geometry")
+    @ScalarFunction("ST_Boundary")
+    @SqlType(NAME)
+    public static Slice stBoundary(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        OGCGeometry bound = geometry.boundary();
+        if (bound.geometryType().equals(MULTI_LINE_STRING) && ((OGCMultiLineString) bound).numGeometries() == 1) {
+            bound = ((OGCMultiLineString) bound).geometryN(0);
+        }
+        return serialize(bound);
+    }
+
+    @Description("Returns a geometry representing the double precision (float8) bounding box of the supplied geometry")
+    @ScalarFunction("ST_Envelope")
+    @SqlType(NAME)
+    public static Slice stEnvelope(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        SpatialReference reference = geometry.getEsriSpatialReference();
+        Envelope envelope = getEnvelope(geometry);
+        return serialize(OGCGeometry.createFromEsriGeometry(envelope, reference));
+    }
+
+    @Description("Returns a geometry that represents that part of geometry A that does not intersect with geometry B")
```

Comment:
nit: references to A and B don't match parameter names left and right

perhaps, copy from the SQL/MM: "Returns the Geometry value that represents the point set difference of two geometries"